src/test/java/org/nixos/idea/NixTestExecutionPolicy.java

```diff
@@ -0,0 +1,20 @@
+package org.nixos.idea;
+
+import com.intellij.testFramework.fixtures.IdeaTestExecutionPolicy;
+
+/**
+ * Test execution policy referenced by the build setup build.gradle.kts.
+ * Tests like {@link org.nixos.idea.lang.LexerFilesTest} use it to locate the test data directory.
+ */
+@SuppressWarnings("unused")
+public class NixTestExecutionPolicy extends IdeaTestExecutionPolicy {
+    @Override
+    protected String getName() {
+        return "Nix";
+    }
+
+    @Override
+    public String getHomePath() {
+        return System.getProperty("plugin.testDataPath");
```

Comment:
That should work, I think. As far as I understand this is only working because it's using the current working dir (i.e. the root project directory) as default.